src/background/modules/ContextMenu.js

```diff
@@ -163,29 +164,59 @@ async function addMenuItems(menuItems, unicodeFontSettings = lastCachedUnicodeFo
             continue;
         }
 
-        const translatedMenuText = browser.i18n.getMessage(transformationId);
-        let textToBeTransformed = translatedMenuText;
-        if (unicodeFontSettings.livePreview && exampleText) {
-            textToBeTransformed = exampleText;
-        }
-        const transformedText = UnicodeTransformationHandler.transformText(textToBeTransformed, transformationId);
-
-        let menuText = transformedText;
-        if (unicodeFontSettings.showReadableText) {
-            menuText = browser.i18n.getMessage("menuReadableTextWrapper", [translatedMenuText, transformedText]);
-        }
-        menuText = menuText.replaceAll("&", "&&");
-
-        if (menuIsShown) {
-            menus.update(transformationId, {
-                title: menuText
-            });
+        if (unicodeFontSettings.nested && amenuItems.length > 1) {
+            const translatedMenuText = browser.i18n.getMessage(transformationId);
+            if (menuIsShown) {
+                menus.update(transformationId, {
+                    title: translatedMenuText
+                });
+            } else {
+                await menus.create({
+                    id: transformationId,
+                    title: translatedMenuText,
+                    contexts: ["editable"]
+                });
+            }
+            for (const atransformationId of amenuItems) {
+                const translatedMenuText = browser.i18n.getMessage(atransformationId);
+                const textToBeTransformed = unicodeFontSettings.livePreview && exampleText ? exampleText : translatedMenuText;
+                const transformedText = UnicodeTransformationHandler.transformText(textToBeTransformed, atransformationId);
+
+                let menuText = unicodeFontSettings.showReadableText ? browser.i18n.getMessage("menuReadableTextWrapper", [translatedMenuText, transformedText]) : transformedText;
+                menuText = menuText.replaceAll("&", "&&");
+                if (menuIsShown) {
+                    menus.update(atransformationId, {
+                        title: menuText
+                    });
+                } else {
+                    await menus.create({
+                        id: atransformationId,
+                        parentId: transformationId,
+                        title: menuText,
+                        contexts: ["editable"]
+                    });
+                }
+            }
         } else {
-            await menus.create({
-                id: transformationId,
-                title: menuText,
-                contexts: ["editable"]
-            });
+            for (const atransformationId of amenuItems) {
```

Comment:
OK, I will remove the `a` prefixes.

Regarding refactoring, I am open to suggestions. I doubt there would ever be multiple levels of nesting, as it would then make it too difficult for users to use.